File: python-programs/guess_the_number.py
from random import randint

print("Guess the Number from 0 to 9!")

target = randint(0, 9)
limit = 5

while limit > 0:
    try:
        guess = int(input("Guess: "))
        if guess not in range(0, 10):
            print("Guess 0 to 9 only.")
            continue
    except ValueError:
        print("Invalid input")
        continue

    if guess == target:
        print(f"Correct! It's {target}")
        break

    limit -= 1
    if limit != 0:
        print("Too high!" if guess > target else "Too low!")
        print(f"{limit} guess remaining.")

else:
    print(f"Wrong! It's {target}.")

File: java-programs/GuessTheNumber.java
package Exercises;

import java.util.Random;
import java.util.Scanner;

public class GuessTheNumber {

	public static void main(String[] args) {

		Random random = new Random();
		Scanner scanner = new Scanner(System.in);

		int limit = 5;
		int number = random.nextInt(10);
		int guess;

		while (limit > 0) {
			System.out.print("Guess: ");
			guess = scanner.nextInt();

			if (guess > number) {
				System.out.println("Too high!");
			} else if (guess < number) {
				System.out.println("Too low!");
			} else {
				break;
			}
			limit--;
		}

		if (limit == 0) {
			System.out.println("You're out of guess. Try again.");
		} else {
			System.out.println("Correct!");
		}

		scanner.close();

	}

}
